### PR TITLE
JAMES-3995 Add a max number of items for EmailGet full reads

### PR DESCRIPTION
**Pull request by vttranlina**
Continue of pr https://github.com/apache/james-project/pull/2014

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala [hunk 1]
```diff
@@ -160,8 +161,7 @@ class EmailGetMethod @Inject() (readerFactory: EmailViewReaderFactory,
       .reduceOption(ReadLevel.combine)
       .getOrElse(MetadataReadLevel)
 
-    // TODO make this configurable
-    if (readLevel.equals(FullReadLevel) && ids.size > 5) {
+    if (readLevel.equals(FullReadLevel) && configuration.jmapEmailGetFullMaxSize.isDefined && ids.size > configuration.jmapEmailGetFullMaxSize.get) {
       SFlux.error(new RuntimeException(s"Too many items in an email read at level FULL. Got ${ids.size} items instead of maximum 5."))
```
Comment (quantranhong1999) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala:165]: `maximum 5` -> should be dynamic now

**Review by quantranhong1999 — APPROVED** [review]

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala [hunk 2]
```diff
@@ -160,9 +161,8 @@ class EmailGetMethod @Inject() (readerFactory: EmailViewReaderFactory,
       .reduceOption(ReadLevel.combine)
       .getOrElse(MetadataReadLevel)
 
-    // TODO make this configurable
-    if (readLevel.equals(FullReadLevel) && ids.size > 5) {
-      SFlux.error(new RuntimeException(s"Too many items in an email read at level FULL. Got ${ids.size} items instead of maximum 5."))
+    if (readLevel.equals(FullReadLevel) && configuration.jmapEmailGetFullMaxSize.isDefined && ids.size > configuration.jmapEmailGetFullMaxSize.get) {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala:164]: Extract this out to a method with a meaningful name?

Comment (Arsnael) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala:164]: is `configuration.jmapEmailGetFullMaxSize.isDefined` necessary here? As you seem to to allocate it with a value by default if it is not defined in the conf?

**Comment by Arsnael** [comment]
Also can you put JAMES-3995 JIRA ticket number to the commits? I think it fits on optimizing Email/get :)

**Comment by vttranlina** [comment]
> Also can you put JAMES-3995 JIRA ticket number to the commits? I think it fits on optimizing Email/get :)

yes, I will do it next time (when rebase & squash)

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**